core/src/main/java/org/opensearch/sql/analysis/Analyzer.java

```diff
@@ -448,6 +454,111 @@ public LogicalPlan visitEval(Eval node, AnalysisContext context) {
+   *
+   * <p>Input Data:
+   *
+   * <pre>
+   * [
+   *   struct: {
+   *     integer: 0,
+   *     nested_struct: { string: "value" }
+   *   }
+   * ]
+   * </pre>
+   *
+   * Query 1: <code>flatten struct</code>
+   *
+   * <pre>
+   * [
+   *   struct: {
+   *     integer: 0,
+   *     nested_struct: { string: "value" }
+   *   },
+   *   integer: 0,
+   *   nested_struct: { string: "value" }
+   * ]
+   * </pre>
+   *
+   * Query 2: <code>flatten struct.nested_struct</code>
+   *
+   * <pre>
+   * [
+   *   struct: {
+   *     integer: 0,
+   *     nested_struct: { string: "value" },
+   *     string: "value"
+   *   }
+   * ]
+   * </pre>
+   */
+  @Override
+  public LogicalPlan visitFlatten(Flatten node, AnalysisContext context) {
+    LogicalPlan child = node.getChild().getFirst().accept(this, context);
+
+    ReferenceExpression fieldExpr =
+        (ReferenceExpression) expressionAnalyzer.analyze(node.getField(), context);
+    String fieldName = fieldExpr.getAttr();
+
+    // [A] Determine fields to add
+    // ---------------------------
+
+    // Iterate over all the fields defined in the type environment. Find all those that are
+    // descended from field that is being flattened, and determine the new paths to add. When
+    // determining the new paths, we need to preserve the portion of the path corresponding to the
+    // flattened field's parent, if one exists, in order to support flattening nested structs.
+
+    Map<String, ExprType> addFieldsMap = new HashMap<>();
+
+    TypeEnvironment env = context.peek();
+    Map<String, ExprType> fieldsMap = env.lookupAllTupleFields(Namespace.FIELD_NAME);
+
+    final String fieldDescendantPath = fieldName + PathUtils.SEPARATOR;
+    final String fieldParentPath =
+        fieldName.contains(PathUtils.SEPARATOR)
+            ? fieldName.substring(0, fieldName.lastIndexOf(PathUtils.SEPARATOR))
+            : "";
+
+    for (String path : fieldsMap.keySet()) {
+
+      // Verify that the path is descended from the flattened field.
+      if (!path.startsWith(fieldDescendantPath)) {
+        continue;
+      }
+
+      // Build the new path.
+      String newPath = path.substring(fieldDescendantPath.length());
+      if (!fieldParentPath.isEmpty()) {
+        newPath = String.join(PathUtils.SEPARATOR, fieldParentPath, newPath);
+      }
+
+      ExprType type = fieldsMap.get(path);
+      addFieldsMap.put(newPath, type);
+    }
+
+    // [B] Add new fields to type environment
+    // --------------------------------------
+
+    for (Map.Entry<String, ExprType> entry : addFieldsMap.entrySet()) {
+      String name = entry.getKey();
+      ExprType type = entry.getValue();
+
+      // Verify that new field does not overwrite an existing field.
+      if (fieldsMap.containsKey(name)) {
+        throw new IllegalArgumentException(
+            StringUtils.format("Flatten command cannot overwrite field '%s'", name));
+      }
+
+      env.define(DSL.ref(name, type));
```

Comment:
you don't need to save entry.getValue() to a variable that is only used once. 
and then you aren't needlessly accessing the variable if it throws
```suggestion
      env.define(DSL.ref(name, entry.getValue()));
```

Reply:
Done! ✅